---
title: C28105
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: c0906c1e876a13049dfe1d1e7727c22528f8e8ed
ms.sourcegitcommit: 94b3a052fb1229c7e7f8804b09c1d403385c7630
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 04/23/2019
ms.locfileid: "62570387"
---
# <a name="c28105"></a>C28105
警告 C28105:由于异常正在泄漏资源

 指定的资源不会释放时引发的异常。 由路径指定的语句将引发异常。 此警告是类似于警告[C28103](../code-quality/c28103.md)，只不过在此示例涉及异常。

## <a name="example"></a>示例
 下面的代码示例将生成此警告：

```
res = KeSaveFloatingPointState(buffer);

res = AllocateResource(Resource);
char *p2 = new char[10]; // could throw

delete[] p2;
FreeResource(Resource)
```

 下面的代码示例可避免此警告：

```
res = AllocateResource(Resource);
char *p2;

try {
    p2 = new char[10];
} catch (std::bad_alloc *e) {
    // just handle the throw
    ;
}
FreeResource(Resource)
```